RecoPixelVertexing/PixelVertexFinding/plugins/gpuVertexFinderImpl.h

```diff
@@ -102,7 +102,7 @@ namespace gpuVertexFinder {
 #endif  // PIXVERTEX_DEBUG_PRODUCE
     ZVertexHeterogeneous vertices(cms::cuda::make_device_unique<ZVertexSoA>(stream));
 #else
-  ZVertexHeterogeneous Producer::make(TkSoA const* tksoa, float ptMin) const {
+  inline ZVertexHeterogeneous Producer::make(TkSoA const* tksoa, float ptMin) const {
```

Comment:
This one should be without the `inline` because it is effectively a source file, whose compiler is dictated by the file extension by the `.cc`/`.cu` files that include this header.
```suggestion
  ZVertexHeterogeneous Producer::make(TkSoA const* tksoa, float ptMin) const {
```

Reply:
Maybe we should rename this and similar files from `.h` to something else ?
And/or, find an other way to compile it multiple times with different compiler and compiler flags ?

Reply:
> Maybe we should rename this and similar files from `.h` to something else ?

That would indeed be a simple way forward.

> And/or, find an other way to compile it multiple times with different compiler and compiler flags ?

With a portability technology (or at least for Kokkos or Alpaka) we'd need such a solution anyway. Then the question is whether we want to make such an exercise already with CUDA or not (I don't have a clear feeling).

Reply:
> With a portability technology (or at least for Kokkos or Alpaka) we'd need such a solution anyway. Then the question is whether we want to make such an exercise already with CUDA or not (I don't have a clear feeling).

I'd say "yes, why now, if we have the time to do it now"... but I don't know if we do :-/

Reply:
do not rename file with extensions LXR or DXR do not understand, please. already .icc or similar are a problem.

Reply:
can we tell LXR and DXR to parse some more extensions ?

Reply:
wonder why not be done till now...

Reply:
I am not aware of LXR/DXR issue. Looks lioke LXR is properly indexing .icc file https://cmssdt.cern.ch/lxr/source/DataFormats/RPCDigi/interface/RPCAMCLinkCounters.icc . 

DXR is also indexing .icc files ( https://cmssdt.cern.ch/dxr/CMSSW/search?q=%2Btype-ref%3ARPCAMCLinkCounters ) but issue is with htmlifing the .icc file.  I can patch our DXR to treat `.icc` files as c++ sources files when htmlifier

Reply:
From the core meeting: for now we could rename this kind of header to `foo.cc` file and then in `foo.cu` file do `#include "foo.cc"`.

Reply:
anything that CORE proposes is fine with me

Reply:
OK, so we do
```bash
git rm -f RecoPixelVertexing/PixelVertexFinding/plugins/gpuVertexFinder.cc
git mv RecoPixelVertexing/PixelVertexFinding/plugins/gpuVertexFinderImpl.h RecoPixelVertexing/PixelVertexFinding/plugins/gpuVertexFinder.cc 
echo '#include "gpuVertexFinder.cc"' > RecoPixelVertexing/PixelVertexFinding/plugins/gpuVertexFinder.cu
git add RecoPixelVertexing/PixelVertexFinding/plugins/gpuVertexFinder.cu
```

Reply:
> Fixed in #33428.

Thanks Andrea!